program/rust/src/rust_oracle.rs

```diff
@@ -169,6 +171,69 @@ pub fn add_price(
     Ok(SUCCESS)
 }
 
+pub fn init_price(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> OracleResult {
+    let cmd_args = load::<cmd_init_price_t>(instruction_data)?;
+
+    check_exponent_range(cmd_args.expo_)?;
+
+    let [funding_account, price_account] = match accounts {
+        [x, y] => Ok([x, y]),
+        _ => Err(ProgramError::InvalidArgument),
+    }?;
+
+    check_valid_funding_account(funding_account)?;
+    check_valid_signable_account(program_id, price_account, size_of::<pc_price_t>())?;
+
+    let mut price_data = load_checked::<pc_price_t>(price_account, cmd_args.ver_)?;
+    pyth_assert(
+        price_data.ptype_ == cmd_args.ptype_,
+        ProgramError::InvalidArgument,
+    )?;
+
+    price_data.expo_ = cmd_args.expo_;
+
+    price_data.last_slot_ = 0;
+    price_data.valid_slot_ = 0;
+    price_data.agg_.pub_slot_ = 0;
+    price_data.prev_slot_ = 0;
+    price_data.prev_price_ = 0;
+    price_data.prev_conf_ = 0;
+    price_data.prev_timestamp_ = 0;
+    sol_memset(
+        bytes_of_mut(&mut price_data.twap_),
+        0,
+        size_of::<pc_ema_t>(),
+    );
+    sol_memset(
+        bytes_of_mut(&mut price_data.twac_),
+        0,
+        size_of::<pc_ema_t>(),
+    );
+    sol_memset(
+        bytes_of_mut(&mut price_data.agg_),
+        0,
+        size_of::<pc_price_info_t>(),
+    );
+    for i in 0..(price_data.comp_.len() as usize) {
```

Comment:
Ok, so the C code uses `price_data.num_` as the upper range. I think you're using the size of the array. In that case what about using `PC_COMP_SIZE`

Reply:
right. i decided to go with len() because it seemed less brittle than `num_`. I think it's more explicit to use `len()` than PC_COMP_SIZE (as in, it's clear to the reader of the code what this does, without them having to look up the definition of PC_COMP_SIZE)